ui/tests/acceptance/auth-test.js

```diff
@@ -193,4 +195,18 @@ module('Acceptance | auth', function (hooks) {
     await fillIn(AUTH_FORM.method, 'token');
     await click('[data-test-auth-submit]');
   });
+
+  test('it does not call renew-self after successful login with non-renewable token', async function (assert) {
+    assert.expect(1);
+
+    this.server.post('/auth/token/renew-self', function () {
+      assert.notOk(true, 'should not call renew-self directly after logging in');
```

Comment:
Noelle had a great suggestion a while ago to throw errors from network requests we didn't want to make, instead of relying on the assertion count: 

https://github.com/hashicorp/vault/blob/1e8eefade13acf0168a12dfbf7af4ef526f3a757/ui/tests/integration/components/dashboard/overview-test.js#L105-L108